tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+	"compilerOptions": {
+		"allowJs": true,
+		"checkJs": true,
+		"noEmit": true,
+		"strict": true,
+		"allowImportingTsExtensions": true,
+		"lib": ["dom", "es2023"],
+		"module": "nodenext"
```

Comment:
🤔 Are `allowImportingTsExtensions` and `module` necessary?

That's a genuine question: My JSDoc-typed projects exclusively use Deno for type checking, so I'm never entirely sure whether regular TypeScript works exactly the same way.

Relatedly, I've long been wondering whether TypeScript configuration should include [`isolatedModules`](https://www.typescriptlang.org/tsconfig/#isolatedModules) and now [`erasableSyntaxOnly`](https://www.typescriptlang.org/tsconfig/#erasableSyntaxOnly) (though the latter should only be relevant for `.ts` source code anyway).

Basically, I'm perennially insecure about the non-obvious nuances of TypeScript configuration (`module` is a good example of that), so I obsessively try to reduce it to the bare minimum.

Reply:
* `allowImportingTsExtensions`: If I remove it, TS fails :sweat_smile: From my understanding this is because we are importing ts files from the comments of JS files.
* module option: Not adding it leads to the error `Module '"node:path"' can only be default-imported using the 'esModuleInterop' flag`
* Activated the two options :+1:

Reply:
Roger, thanks. (That suggests using Deno for type checking is not a great idea because obscure abstractions vs. using the respective tool directly.)